### PR TITLE
REST file import endpoint

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: cmd/lotus/rpc.go [hunk 1]
```diff
@@ -44,3 +55,30 @@ func serveRPC(a api.FullNode, stop node.StopFunc, addr string) error {
...
+			w.WriteHeader(404)
+			return
+		}
+		if !api.HasPerm(r.Context(), api.PermWrite) {
+			w.WriteHeader(401)
+			json.NewEncoder(w).Encode(struct{ Error string }{"unauthorized: missing write permission"})
+			return
+		}
+
+		c, err := a.(*impl.FullNodeAPI).ClientImportLocal(r.Context(), r.Body)
```
Comment (whyrusleeping) [inline cmd/lotus/rpc.go]: why assert here?

Comment (magik6k) [reply cmd/lotus/rpc.go]: `ClientImportLocal` is not exposed on the api interface / jsonrpc (because `io.Reader`)

File: docs/API.md [hunk 2]
```diff
@@ -4,8 +4,13 @@ TODO: make this into a nicer doc
 
 By default `127.0.0.1:1234` - daemon stores the api endpoint multiaddr in `~/.lotus/api`
 
-* `http://[api:port]/rpc/v0` - jsonrpc http endpoint
-* `ws://[api:port]/rpc/v0` - jsonrpc websocket endpoint
+* `http://[api:port]/rpc/v0` - JsonRPC http endpoint
+* `ws://[api:port]/rpc/v0` - JsonRPC websocket endpoint
+* `PUT http://[api:port]rest/v0/import` - import file to the node repo
```
Comment (whyrusleeping) [inline docs/API.md]:
```suggestion
* `PUT http://[api:port]/rest/v0/import` - import file to the node repo
```

File: docs/API.md [hunk 3]
```diff
@@ -4,8 +4,13 @@ TODO: make this into a nicer doc
...
+  * Requires write permission
```
Comment (whyrusleeping) [inline docs/API.md]:
```suggestion
  * Requires admin permission
```

File: cmd/lotus/rpc.go (hunk 1 above)
Comment (Kubuxu) [reply cmd/lotus/rpc.go]: IMO, it would be better to do that cast in L37, instead of here.

File: docs/API.md (hunk 3 above)
Comment (magik6k) [reply docs/API.md]: No, that one only requires `write` - https://github.com/filecoin-project/lotus/pull/464/files#diff-3049d9a11322263539c11d573869b5baR65

(and I've changed the other one as you could technically import the admin token from the repo with that method..)

Comment (whyrusleeping) [reply docs/API.md]: Ah, got it

**Pull request merged**